### PR TITLE
fix: Grant access to TestPreview and Previewer actions for TestAuthors

### PR DESCRIPTION
**Pull request by hectoras**
**Associated Jira issue:** [ADF-439](https://oat-sa.atlassian.net/browse/ADF-439)

Changes in this PR grant access to item and test preview actions for test authors.

### CODE REVIEW

**Review by gabrielfs7 — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: migrations/Version202204201509000545_taoQtiTestPreviewer.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    {
+        return 'Assign permissions to Test Author role';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $setRolesAccess = $this->propagate(new SetRolesAccess());
+        $setRolesAccess([
+            '--' . SetRolesAccess::OPTION_CONFIG, self::CONFIG,
+        ]);
```
Comment (shpran) [inline migrations/Version202204201509000545_taoQtiTestPreviewer.php:68]: Please, use `$this->runAction()` instead.

File: migrations/Version202204201509000545_taoQtiTestPreviewer.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        ]);        
+    }
+
+    public function down(Schema $schema): void
+    {
+        $setRolesAccess = $this->propagate(new SetRolesAccess());
+        $setRolesAccess([
+            '--' . SetRolesAccess::OPTION_REVOKE,
+            '--' . SetRolesAccess::OPTION_CONFIG, self::CONFIG,
+        ]);
```
Comment (shpran) [inline migrations/Version202204201509000545_taoQtiTestPreviewer.php:77]: Please, use `$this->runAction()` instead.

**Review by shpran — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [ ] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

File: migrations/Version202204201509000545_taoQtiTestPreviewer.php (hunk 2 above)
Comment (hectoras) [reply migrations/Version202204201509000545_taoQtiTestPreviewer.php]: Changed in e48d50da702bb5f60e1bb785879e54d4d1ce3516 (today I learned :D) Thanks!

File: migrations/Version202204201509000545_taoQtiTestPreviewer.php (hunk 1 above)
Comment (hectoras) [reply migrations/Version202204201509000545_taoQtiTestPreviewer.php]: Changed in e48d50da702bb5f60e1bb785879e54d4d1ce3516 (today I learned :D) Thanks!

**Comment by hectoras** [comment]
Released as [v3.5.2](https://github.com/oat-sa/extension-tao-testqti-previewer/releases/tag/v3.5.2).

**Comment by hectoras** [comment]
Backported to tao-communityMarch release as [2022.03.15](https://github.com/oat-sa/tao-community/releases/tag/2022.03.15).

**Comment by AntonNeviarouski** [comment]
QA verified

**Pull request merged**